sp/src/game/server/ez2/npc_pitdrone.cpp

```diff
@@ -210,6 +229,57 @@ int CNPC_PitDrone::RangeAttack1Conditions( float flDot, float flDist )
 	return(BaseClass::RangeAttack1Conditions( flDot, flDist ));
 }
 
+//=========================================================
+// Start task - selects the correct activity and performs
+// any necessary calculations to start the next task on the
+// schedule. 
+//
+// Overridden for bullsquids to play specific activities
+//=========================================================
+void CNPC_PitDrone::StartTask( const Task_t *pTask )
+{
+	CGravityVortexController *pVortex;
+
+	switch (pTask->iTask)
+	{
+	case TASK_PREDATOR_SPAWN:
+		// TODO - This should probably have an animation
+
+		// Play a sound
+		EmitSound( "NPC_PitDrone.SummonBackup" );
+
+		// Create a portal
+		pVortex = CGravityVortexController::Create( GetAbsOrigin(), 0.0f, 0.0f, 0.0f, this );
+		pVortex->AddContext( "owner_classname:npc_pitdrone" );
```

Comment:
I can't find the code where this is used. Is it left over from a previous implementation?

Reply:
This is used to distinguish pitdrone created singularities from any others. I use it in the response scripts in the mod itself.